graphql/resolvers.go

```diff
@@ -254,6 +257,58 @@ func (r *queryResolver) TaskTests(ctx context.Context, taskID string, sortCatego
 	return testPointers, nil
 }
 
+func getGroupedFiles(ctx context.Context, name string, taskID string, execution int) (*GroupedFiles, error) {
+	taskFiles, err := artifact.GetAllArtifacts([]artifact.TaskIDAndExecution{{TaskID: taskID, Execution: execution}})
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, err.Error())
+	}
+	strippedFiles := artifact.StripHiddenFiles(taskFiles, gimlet.GetUser(ctx))
+	apiFileList := []*restModel.APIFile{}
+	for _, file := range strippedFiles {
+		apiFile := restModel.APIFile{}
+		err := apiFile.BuildFromService(file)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, err.Error())
+		}
+		apiFileList = append(apiFileList, &apiFile)
+	}
+	return &GroupedFiles{Name: &name, Files: apiFileList}, nil
+}
+
+func (r *queryResolver) TestFiles(ctx context.Context, taskID string) ([]*GroupedFiles, error) {
+	t, err := task.FindOneId(taskID)
+	if t.OldTaskId != "" {
+		t, err = task.FindOneId(t.OldTaskId)
+		if err != nil {
+			return nil, ResourceNotFound.Send(ctx, err.Error())
+		}
+	}
+	groupedFilesList := []*GroupedFiles{}
+	if t.DisplayOnly {
+		for _, execTaskID := range t.ExecutionTasks {
+			execTask, err := task.FindOne(task.ById(execTaskID))
+			if err != nil {
+				return nil, ResourceNotFound.Send(ctx, err.Error())
+			}
+			if execTask == nil {
+				continue
+			}
+			groupedFiles, err := getGroupedFiles(ctx, execTask.DisplayName, execTaskID, t.Execution)
+			if err != nil {
+				return nil, err
+			}
+			groupedFilesList = append(groupedFilesList, groupedFiles)
+		}
+	} else {
+		groupedFiles, err := getGroupedFiles(ctx, t.DisplayName, taskID, t.Execution)
+		if err != nil {
+			return nil, err
+		}
+		groupedFilesList = append(groupedFilesList, groupedFiles)
```

Comment:
i see some conditional logic here and it makes me wonder if different results will be produced from this resolver based on if t.DisplayOnly or not. are the results the same or different? if different should there be a test case for each outcome?

Reply:
`test-files-query-exec.graphql` tests when t.DisplayOnly is false and `test-files-query-display.graphql` tests when t.DisplayOnly is true